toolkit/tools/pkg/imagecustomizerlib/imagecustomizer.go

```diff
@@ -930,3 +936,28 @@ func CheckEnvironmentVars() error {
 
 	return nil
 }
+
+// validateTargetOs checks if the current distro/version is supported and has the required preview features enabled
+func validateTargetOs(ctx context.Context, buildDir string, buildImageFile string, config *imagecustomizerapi.Config) error {
+	existingImageConnection, _, _, _, err := connectToExistingImage(ctx, buildImageFile, buildDir, "imageroot", false,
+		true, false, false)
```

Comment:
nit: would be good to add inline comments to the 'false/true' arguments to clarify what they mean for the reader. For example '... false /*read-only*/, ...'